### PR TITLE
[WIP] Setup prettier with ESLint

### PR DESCRIPTION
**Pull request by Slashgear**

### CODE REVIEW

**Comment by Swiip** [comment]
I checked your PR, first, it should be cleaned up as almost all the code is modified with mostly space indents and some `.idea` files which should not be here at all. For the moment I which the PR does not touch the code, we'll apply prettier at the end when everything else is ok.

Passed that, I pushed some changes. I used the prettier cli from the root node_modules like I did for ESLint. I also removed prettier from the global link task. Indeed, prettier task will change the source code... CI can't be allowed to touch the source code. ESLint will check that the code is conform but it will be the devs which have to run prettier. If it causes pb, we'll add prettier as a pre-commit hook.

Finally the tabs pb. I tried `prettier-with-tabs` but it's a violent fork and it will cause us pbs. It already the case as it's not possible to configure `eslint-plugin-prettier` to use it or even just allow tabs.

To sum up:
- Clean changes in the PR
- Choose between:
  - Switch to spaces
  - Hope that https://github.com/prettier/prettier/issues/908 will be merged

**Comment by rhengles** [comment]
Violent? Lol 😄 

If you could give detail in what "pbs" it could cause, that would be helpful 😉

**Comment by Swiip** [comment]
@rhengles Sorry for the strong word ^^ Did not mean to offend. Right now, clash with `eslint-plugin-prettier` is blocking for us. Hope your PR will be merged.

I don't know how far I'll persist with "my" tabs. I think that if Prettier officially refuse to support it, I could switch to spaces.

**Comment by Swiip** [comment]
Ooops, you asked for details, here there are. We are heavy users of ESLint and we planned to ensure prettier usage with it. To do this, we need `eslint-plugin-prettier` and `eslint-config-prettier`. But these tools seems not to be configurable, not much than prettier at least.

So when we apply `prettier-with-tabs` ESlint fail on the tabs. I found no way to make ESLint pass without stopping all ESLint checks which is quite not acceptable...

**Comment by rhengles** [comment]
I understand why many people don't want to run from a fork - as I said myself on prettier/prettier#908, I hated when StandardJS required a fork to insert semicolons.

Thank you, if prettier/prettier#908 is rejected, I will make sure to create a `eslint-plugin-prettier-with-tabs`...

But I only want to say this - `prettier-with-tabs` is a superset of `prettier` and will always pass the same tests from `prettier` by default, it will only be different if you configure it.

**Comment by Swiip** [comment]
Prettier 1.0 with tabs included is released 🎉 

Waiting for the big PR of Chloé and then we can re-open this !

**Comment by Swiip** [comment]
Closing in favor of #61 more recent based on prettier 1.4 supporting tabs
